network/src/sync/blocks.rs

```diff
@@ -162,42 +143,19 @@ impl<S: Storage + Send + std::marker::Sync + 'static> Node<S> {
     pub(crate) async fn received_get_sync(
         &self,
         remote_address: SocketAddr,
-        block_locator_hashes: Vec<BlockHeaderHash>,
+        block_locator_hashes: Vec<Digest>,
     ) -> Result<(), NetworkError> {
-        let sync = {
-            let storage = self.expect_sync().storage();
-
-            let latest_shared_hash = storage.get_latest_shared_hash(block_locator_hashes)?;
-            let current_height = storage.get_current_block_height();
-
-            if let Ok(height) = storage.get_block_number(&latest_shared_hash) {
-                if height < current_height {
-                    let mut max_height = current_height;
-
-                    // if the requester is behind more than MAX_BLOCK_SYNC_COUNT blocks
-                    if current_height > height + crate::MAX_BLOCK_SYNC_COUNT {
-                        // send no more than MAX_BLOCK_SYNC_COUNT
-                        max_height = height + crate::MAX_BLOCK_SYNC_COUNT;
-                    }
-
-                    let mut block_hashes = Vec::with_capacity((max_height - height) as usize);
-
-                    for block_num in height + 1..=max_height {
-                        block_hashes.push(storage.get_block_hash(block_num)?);
-                    }
-
-                    // send block hashes to requester
-                    block_hashes
-                } else {
-                    vec![]
-                }
-            } else {
-                vec![]
-            }
-        };
+        let sync_hashes = self
+            .storage
+            .find_sync_blocks(&block_locator_hashes[..], crate::MAX_BLOCK_SYNC_COUNT as usize)
+            .await?
+            .into_iter()
+            .map(|x| x.bytes::<32>().map(BlockHeaderHash))
```

Comment:
We might want to introduce a `impl TryFrom<Digest> for BlockHeaderHash` (or `TryInto` the other way round).

Reply:
`TryInto` is usually messy to use -- and keeping `BlockHeaderHash` around is a temporary measure